riscv/csrs.cc

```diff
@@ -504,9 +507,14 @@ bool misa_csr_t::unlogged_write(const reg_t val) noexcept {
   if (!(val & (1L << ('C' - 'A'))) && (state->pc & 2))
     return false;
 
-  const bool val_supports_f = val & (1L << ('F' - 'A'));
-  const reg_t val_without_d = val & ~(1L << ('D' - 'A'));
-  const reg_t adjusted_val = val_supports_f ? val : val_without_d;
+  reg_t adjusted_val = val;
+  if (!(adjusted_val & (1L << ('F' - 'A')))) adjusted_val &= ~(1L << ('D' - 'A'));
+  if (!(adjusted_val & (1L << ('D' - 'A')))) adjusted_val &= ~((1L << ('V' - 'A')) | (1L << ('Q' - 'A')));
+  if (adjusted_val & ((1L << ('V' - 'A')) | (1L << ('Q' - 'A')))) adjusted_val |= (1L << ('D' - 'A'));
+  if (adjusted_val & (1L << ('D' - 'A'))) adjusted_val |= (1L << ('F' - 'A'));
```

Comment:
This logic doesn't make sense. If D=0, line 512 clears V & Q, so line 513 will never be true. Same for line 514 and 517.

Reply:
Sorry for tagging.
I already realized my stupidity

Reply:
I re-read spec and realised that there is no logic of re-enabling implied extensions.